pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -138,8 +138,26 @@ func (c *Controller) sync() error {
 	serviceCIDR, err := c.serviceCIDRLister.Get(DefaultServiceCIDRName)
 	// if exists
 	if err == nil {
-		c.syncStatus(serviceCIDR)
-		return nil
+		// single to dual stack upgrade
+		if len(c.cidrs) == 2 && len(serviceCIDR.Spec.CIDRs) == 1 && c.cidrs[0] == serviceCIDR.Spec.CIDRs[0] {
+			klog.Infof("Updating default ServiceCIDR with CIDRs: %v", c.cidrs)
```

Comment:
maybe explicitly log the before / after and make it clear this is a single-stack → dual-stack expansion?

```suggestion
			klog.Infof("Updating default ServiceCIDR from single-stack (%v) to dual-stack (%v)", serviceCIDR.Spec.CIDRs, c.cidrs)
```